testdata/helm/memcached-operator/Makefile

```diff
@@ -113,3 +120,49 @@ bundle: kustomize
 .PHONY: bundle-build
 bundle-build:
 	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
+
+# Push the bundle image.
+.PHONY: bundle-push
+bundle-push:
+	$(MAKE) docker-push IMG=$(BUNDLE_IMG)
+
+# Download opm locally if necessary.
+.PHONY: opm
+OPM = ./bin/opm
+opm:
+ifeq (,$(wildcard $(OPM)))
+ifeq (,$(shell which opm 2>/dev/null))
+	@{ \
+	set -e ;\
+	mkdir -p $(dir $(OPM)) ;\
+	curl -sSLo $(OPM) https://github.com/operator-framework/operator-registry/releases/download/v1.15.1/$(OS)-$(ARCH)-opm ;\
+	chmod +x $(OPM) ;\
+	}
+else
+OPM = $(shell which opm)
+endif
+endif
+
+# A comma-separated list of bundle images (e.g. make catalog-build BUNDLE_IMGS=example.com/operator-bundle:v0.1.0,example.com/operator-bundle:v0.2.0).
+# These images MUST exist in a registry and be pull-able.
+BUNDLE_IMGS ?= $(BUNDLE_IMG)
+
+# The image tag given to the resulting catalog image (e.g. make catalog-build CATALOG_IMG=example.com/operator-catalog:v0.2.0).
+CATALOG_IMG ?= $(IMAGE_TAG_BASE)-catalog:v$(VERSION)
+
+# Set CATALOG_BASE_IMG to an existing catalog image tag to add $BUNDLE_IMGS to that image.
+ifneq ($(origin CATALOG_BASE_IMG), undefined)
+FROM_INDEX_OPT := --from-index $(CATALOG_BASE_IMG)
+endif
+
+# Build a catalog image by adding bundle images to an empty catalog using the operator package manager tool, 'opm'.
+# This recipe invokes 'opm' in 'semver' bundle add mode. For more information on add modes, see:
+# https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#updating-your-existing-operator
+.PHONY: catalog-build
+catalog-build: opm
+	$(OPM) index add --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS) $(FROM_INDEX_OPT)
```

Comment:
Will there be a case where we will be adding bundles images to a catalog that's not empty? 
In that case, `semver` bundle mode wouldn't work I guess.

Reply:
If a user sets `FROM_INDEX_OPT` then the base catalog won't be empty. Semver should work in either case. Why don't you think it will?

Reply:
I was under the impression that we have to use `Replaces` mode when adding to a non-empty catalog. If Semver works in either case, that's fine.

Reply:
This isn't a problem I'm aware of. Perhaps you're referring to the default add mode in `operator-sdk run bundle` if the default index image base is used?

Reply:
Ah yes, I'm referring to that. It's not applicable in this case then. :)